scio-core/src/main/java/com/spotify/scio/transforms/FutureHandlers.java

```diff
@@ -92,7 +99,7 @@ default void waitForFutures(Iterable<CompletableFuture<V>> futures)
         throws InterruptedException, ExecutionException {
       CompletableFuture[] array =
           StreamSupport.stream(futures.spliterator(), false).toArray(CompletableFuture[]::new);
-      CompletableFuture.allOf(array).get();
```

Comment:
all implementation do not throw in `waitForFutures` but this one. Change to match same behavior